src/tests/collate_tests/test_from_obj.py

```diff
@@ -133,3 +134,14 @@ def test_materialized_from_obj_should_handle_keywords():
     from_obj = FromObj(from_obj="events", name="events", knowledge_date_column="date")
     assert not from_obj.should_materialize()
     assert from_obj.table == "events"
+
+
+def test_materialized_from_obj_maybe_materialize(db_engine_with_events_table):
+    from_obj = FromObj(
+        from_obj="events", 
+        name="myquery",
+        knowledge_date_column='event_date'
+    )
+    from_obj.should_materialize = lambda: True
+    from_obj.maybe_materialize(db_engine_with_events_table)
+    assert table_exists(from_obj.table, db_engine_with_events_table)
```

Comment:
not sure it will be a problem, but is this by any chance from a different branch/PR?